subprojects/model-core/src/main/java/org/gradle/internal/instantiation/AbstractClassGenerator.java

```diff
@@ -90,6 +90,8 @@
  * </ul>
  */
 abstract class AbstractClassGenerator implements ClassGenerator {
+    private static final ImmutableSet<Class<?>> LAZY_PROPERTY_TYPES = ImmutableSet.of(ConfigurableFileCollection.class, ListProperty.class, SetProperty.class, MapProperty.class, RegularFileProperty.class, DirectoryProperty.class, Property.class);
```

Comment:
This should be split one item per line so it's easy to see what gets added/removed later.

Reply:
Done.